src/core/dbus-device.c

```diff
@@ -1,11 +1,49 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
+#include "bus-get-properties.h"
+#include "cgroup-util.h"
 #include "dbus-device.h"
 #include "device.h"
 #include "unit.h"
 
+static BUS_DEFINE_PROPERTY_GET_ENUM(property_get_iocost_ctrl, io_cost_ctrl, IOCostCtrl);
+
+static int property_get_io_cost_qos_scaling_percentage(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        loadavg_t io_cost_qos_scaling_percentage = *(loadavg_t *) userdata;
+        uint32_t result;
+
+        assert(bus);
+        assert(reply);
+
+        result = LOADAVG_INT_SIDE(io_cost_qos_scaling_percentage) * 100 + LOADAVG_DECIMAL_SIDE(io_cost_qos_scaling_percentage);
+        return sd_bus_message_append_basic(reply, 'u', &result);
+}
+
 const sd_bus_vtable bus_device_vtable[] = {
         SD_BUS_VTABLE_START(0),
         SD_BUS_PROPERTY("SysFSPath", "s", NULL, offsetof(Device, sysfs), SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
+        SD_BUS_PROPERTY("IOCostQOSEnabled", "b", NULL, offsetof(Device, io_cost_qos.enabled), SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
+        SD_BUS_PROPERTY("IOCostQOSCtrl", "s", property_get_iocost_ctrl, offsetof(Device, io_cost_qos.ctrl), SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
```

Comment:
What about merging IOCostQOSEnabled= and IOCostQOSCtrl= into one directive?
IOW is there any semantical difference between `enabled=0,ctrl=auto` and `enabled=0,ctrl=user`?

Reply:
These are semantically the same, but these values are meant to correspond directly to what is in the documentation here: https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#io-interface-files. 

I believe it would be confusing for the user if we merged these into one directive, it would deviate from the original interface.